pkg/helper/log/pipeline_reader.go

```diff
@@ -96,34 +81,33 @@ func (lr *LogReader) readLiveLogs(pr *v1alpha1.PipelineRun) (<-chan log.Log, <-c
 	return logC, errC, nil
 }
 
-func (lr *LogReader) readAvailableLogs(pr *v1alpha1.PipelineRun) (<-chan log.Log, <-chan error, error) {
-	if err := lr.waitUntilAvailable(10); err != nil {
+func (r *Reader) readAvailablePipelineLogs(pr *v1alpha1.PipelineRun) (<-chan Log, <-chan error, error) {
+	if err := r.waitUntilAvailable(10); err != nil {
 		return nil, nil, err
 	}
 
-	tkn := lr.Clients.Tekton
-	pl, err := tkn.TektonV1alpha1().Pipelines(lr.Ns).Get(pr.Spec.PipelineRef.Name, metav1.GetOptions{})
+	tkn := r.clients.Tekton
+	pl, err := tkn.TektonV1alpha1().Pipelines(r.ns).Get(pr.Spec.PipelineRef.Name, metav1.GetOptions{})
 	if err != nil {
-		return nil, nil, fmt.Errorf(err.Error())
+		return nil, nil, err
 	}
 
-	//Sort taskruns, to display the taskrun logs as per pipeline tasks order
+	// Sort taskruns, to display the taskrun logs as per pipeline tasks order
```

Comment:
We need to do sorting for readLivePipelineLogs too.

Reply:
Because ```readLivePipelineLogs``` get logs asynchronous in concurrent goroutine.
Logs are displayed in order from the started pod and this timing is uncontrollable.
if we sort order in ```readLivePipelineLogs``` we need to wait until all task pods will start.
(But it maybe lack of real time... tradeoff...🤔)

Reply:
Ahh... sorry, plase let me considering it.
there is possibility I misunderstanding it, sorry :sweat_smile:

Reply:
I follow what you're saying, and that was my thought here too as far as having to wait for all TaskRuns. I figured I would ask to make sure but no worries if this is something out of the scope of this pr. We can always iterate on it if there isn't a clear solution here.

Reply:
> I follow what you're saying, and that was my thought here too as far as having to wait for all TaskRuns. I figured I would ask to make sure but no worries if this is something out of the scope of this pr. We can always iterate on it if there isn't a clear solution here.

Thank you very much !!
yes, it seems be better addressing this on another PR :pray: